src/useAxiosReducer.ts

```diff
@@ -11,11 +11,11 @@ type Action<Data> =
   | { type: 'REQUEST_SUCCESS'; payload: Data }
   | { type: 'REQUEST_FAILED'; payload: Error };
 
-export const initialState = {
-  data: undefined,
+export const createInitialState = <Data>({ data }: { data?: Data } = {}) => ({
+  data: typeof data === 'undefined' ? undefined : data,
```

Comment:
Could this just be reduced to `data: data`? (+ object shorthand)

Reply:
I changed the signature with my latest push.